File: frontend/src/types/meta/meta.ts
/**
 * Meta domain contracts — aligned with the backend `meta` domain
 * (`GET /`, `GET /health`, `GET /ping`).
 */

/** `GET /` service banner. */
export interface RootData {
  service: string
  version: string
  docs: string
  ping: string
}

/** `GET /health` payload. */
export interface HealthData {
  status: string
  version: string
  uptime_seconds: number
}

/** `GET /ping` payload. */
export interface PingData {
  message: string
  seq: number
  timestamp: string
}

File: backend/app/schemas/meta.py
"""Meta-domain response DTOs (root banner, health, ping)."""

from __future__ import annotations

from pydantic import BaseModel


class RootData(BaseModel):
    service: str
    version: str
    docs: str
    ping: str


class HealthData(BaseModel):
    status: str = "ok"
    version: str
    uptime_seconds: float


class PingData(BaseModel):
    message: str = "pong"
    seq: int
    timestamp: str
